src/NUnitFramework/framework/Constraints/PathConstraint.cs

```diff
@@ -117,7 +116,10 @@ protected bool IsSubPath(string path1, string path2)
                 return false;
 
             // path 2 is longer than path 1: see if initial parts match
-            if (!StringUtil.StringsEqual(path1, path2.Substring(0, length1), caseInsensitive))
+            StringComparison comparisonType = caseInsensitive
```

Comment:
One suggestion: As `SamePathConstraint` and `SamePathOrUnderConstraint` both inherit from `PathConstraint` it may help avoid duplication if we exposed a `protected` property or method here to encapsulate the case-sensitivity check and return a `StringComparison` for use.

Reply:
Hey @stevenaw ! Thanks for the review, I'll make this improvement right away.